examples/coap-server.c

```diff
@@ -2876,9 +2896,9 @@ main(int argc, char **argv) {
        */
       result = coap_io_process( ctx, wait_ms );
```

Comment:
Perhaps a better way is to test for `reload_config` in the main while loop and then call `fill_keystore()` if set, and then unset `reload_config`.  I do not think it safe to `call keystore()` in `handle_sigusr1()`.  It also means that if coap_io_process() also indicates there it did consume some time, other than returning -1, then `call keystore()` can then still take place.

This then means no change to `src/net.c` and no need to special case EINTR returns which could be for any interrupt.

Reply:
> Perhaps a better way is to test for `reload_config` in the main while loop and then call `fill_keystore()` if set, and then unset `reload_config`. I do not think it safe to `call keystore()` in `handle_sigusr1()`. 

Yes, that is what I also thought, especially because this is only for the example server and should not touch the core libcoap logic which is to pass EINTR to the application.